tester/src/test/scala/spinal/tester/scalatest/SpinalSimWishboneSlaveFactoryTester.scala

```diff
@@ -43,55 +46,66 @@ class SpinalSimWishboneSlaveFactoryTester extends SpinalAnyFunSuite{
       for(ref <- List(
         WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 1),
         WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 1),
-        WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 2)
+        WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 2),
+        WishboneTransaction(21 * wordInc, 1),
+        WishboneTransaction(21 * wordInc, 1),
+        WishboneTransaction(21 * wordInc, 42)
       )) {
         scoreboard.pushRef(ref)
       }
 
-      WishboneMonitor(dut.io.bus, dut.clockDomain){ bus =>
-           scoreboard.pushDut(WishboneTransaction.sampleAsMaster(bus))
-      }
+      new WishboneMonitor(dut.io.bus, dut.clockDomain).addResponseCallback(
+        (bus, transaction, we) => {
+          scoreboard.pushDut(transaction.copy(data = bus.DAT_MISO.toBigInt))
+        }
+      )
+
+      // Make sure that all configurations exhibit proper latencies.
```

Comment:
while you do measure the response time, there is not check around here?

Reply:
I checked in the commented out assert version, I can fix that.